conu/apidefs/container.py

```diff
@@ -0,0 +1,234 @@
+
+    def get_ports(self):
+        """
+        get ports specified in container metadata
+
+        :return: list of int
+        """
+        raise NotImplementedError("get_ports method is not implemented")
+
+    def open_connection(self, port=None):
+        """
+        open a TCP connection to service running in the container, if port is None and
+        container exposes only a single port, connect to it, otherwise raise an exception
+
+        :param port: int or None
+        :return: list of int
+        """
+        raise NotImplementedError("open_connection method is not implemented")
+
+    @classmethod
+    def run_via_binary(cls, image, args_builder):
+        """
+        create container using provided image and run it in the background; this method is useful
+        to test real user scenarios when users invoke containers using binary and not an API
+
+        :param image: instance of Image
+        :param args_builder: instance of BinaryArgsBuilder
+        :return: instance of Container
+        """
+        raise NotImplementedError("run_via_binary method is not implemented")
+
+    @classmethod
+    def run_via_api(cls, image, container_params):
+        """
+        create container using provided image and run it in the background
+
+        :param image: instance of Image
+        :param container_params: instance of ContainerParameters
+        :return: instance of Container
+        """
+        raise NotImplementedError("run_via_api method is not implemented")
+
+    @classmethod
+    def create(cls, image, container_params):
+        """
+        create container using provided image
+
+        :param image: instance of Image
+        :param container_params: instance of ContainerParameters
+        :return: instance of Container
+        """
+        raise NotImplementedError("create method is not implemented")
+
+    def start(self):
+        """
+        start current container
+        :return: None
+        """
+        raise NotImplementedError("start method is not implemented")
+
+    def exec(self, command, **kwargs):
+        """
+        TODO: what about parameters?
+
+        :param command: command to execute in the container
+        :param kwargs:
+        :return: ? we need to provide output, exit code and there needs to be a possiblity for
+                  this thing to be async and blocking
+        """
+        raise NotImplementedError("exec method is not implemented")
+
+    def logs(self, follow=False):
+        """
+        get logs from this container
+
+        :param follow: bool, provide iterator if True
+        :return: str or iterator
+        """
+        raise NotImplementedError("logs method is not implemented")
+
+    def stop(self):
+        """
+        stop this container
+
+        :return: None
+        """
+        raise NotImplementedError("stop method is not implemented")
+
+    def rm(self, force=False):
```

Comment:
I think we should also allow removing any mounted volumes here, e.g. the same as `docker rm --volumes`.

Reply:
This is actually a more generic question: what if the functionality you propose is specific to docker? shall we add `**kwargs` here? (or even to all methods?)

Reply:
Good point. I think adding `**kwargs` makes sense.